src/compound_model.js

```diff
@@ -0,0 +1,81 @@
+import {
+    FlowchartEntityMixin,
+    InMemoryEntity,
+    NamedEntityMixin,
+} from "@exabyte-io/code.js/dist/entity";
+import { setNextLinks, setUnitsHead } from "@exabyte-io/code.js/dist/utils";
+import { mix } from "mixwith";
+
+import { UnitModel } from "./unit_model";
+import { UnitModelFactory } from "./unit_model_factory";
+import { buildModelPath } from "./utils";
+
+export class Model extends mix(InMemoryEntity).with(NamedEntityMixin, FlowchartEntityMixin) {
+    static UnitModelFactory = UnitModelFactory;
+
+    static UnitModelCls = UnitModel;
+
+    constructor({ ...config } = {}) {
+        super(config);
+        this._units = Model.instantiateUnits(config.units);
+    }
+
+    get modelPath() {
+        return this.prop("modelPath");
```

Comment:
So do we store the modelPath as a prop or do we generate it dynamically?

Reply:
I think we need the ability to create it dynamically, as the number of units is also dynamic. We could remove `modelPath` as a prop, but so far it is still part of the [schema](https://github.com/Exabyte-io/esse/blob/epic/PNO-72/schema/model_units/unit_base_referable.json) (which can be changed of course).